quantum/matrix_common.c

```diff
@@ -104,15 +120,25 @@ __attribute__((weak)) void matrix_init(void) {
         matrix[i]     = 0;
     }
 
-    debounce_init(MATRIX_ROWS);
+#ifdef SPLIT_KEYBOARD
+    thisHand = isLeftHand ? 0 : (ROWS_PER_HAND);
+    thatHand = ROWS_PER_HAND - thisHand;
+#endif
+
+    debounce_init(ROWS_PER_HAND);
 
     matrix_init_quantum();
 }
 
 __attribute__((weak)) uint8_t matrix_scan(void) {
     bool changed = matrix_scan_custom(raw_matrix);
 
-    debounce(raw_matrix, matrix, MATRIX_ROWS, changed);
+#ifdef SPLIT_KEYBOARD
```

Comment:
I don't love doing this with an #ifdef, but IMO doing something like setting `thisHand` to 0 and passing that for a non-split seems like it would be confusing.